jobs/build.Jenkinsfile

```
@@ -225,26 +225,27 @@ lock(resource: "build-${params.STREAM}") {
         }
 
         // buildfetch previous build info
-        stage('BuildFetch') {
-            if (s3_stream_dir) {
-                pipeutils.shwrapWithAWSBuildUploadCredentials("""
-                cosa buildfetch --arch=${basearch} \
-                    --url s3://${s3_stream_dir}/builds \
-                    --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
-                """)
-                if (parent_version != "") {
-                    // also fetch the parent version; this is used by cosa to do the diff
+        lock(resource: "build-${params.STREAM}") {
```

This will lock any other builds in the `build` stage, we should have a lock dedicated for this fetch operation: 
```suggestion
        lock(resource: "buildfetch") {
```

Since multiple streams jobs can start at the same time let's also have a global lock. It's less than <20s so it's not going to stale the whole pipeline for long.

---

Ok, that makes sense. Thanks!